opm/simulators/utils/ParallelEclipseState.cpp

```diff
@@ -0,0 +1,310 @@
+           serializer.packSize(m_faults) +
+           serializer.packSize(m_title);
+
+}
+
+
+void ParallelEclipseState::pack(std::vector<char>& buffer, int& position,
+                                EclMpiSerializer& serializer) const
+{
+    serializer.pack(m_tables, buffer, position);
+    serializer.pack(m_runspec, buffer, position);
+    serializer.pack(m_eclipseConfig, buffer, position);
+    serializer.pack(m_deckUnitSystem, buffer, position);
+    serializer.pack(m_inputNnc, buffer, position);
+    serializer.pack(m_inputEditNnc, buffer, position);
+    serializer.pack(m_simulationConfig, buffer, position);
+    serializer.pack(m_transMult, buffer, position);
+    serializer.pack(m_faults, buffer, position);
+    serializer.pack(m_title, buffer, position);
+}
+
+
+void ParallelEclipseState::unpack(std::vector<char>& buffer, int& position,
+                                  EclMpiSerializer& serializer)
+{
+    serializer.unpack(m_tables, buffer, position);
+    serializer.unpack(m_runspec, buffer, position);
+    serializer.unpack(m_eclipseConfig, buffer, position);
+    serializer.unpack(m_deckUnitSystem, buffer, position);
+    serializer.unpack(m_inputNnc, buffer, position);
+    serializer.unpack(m_inputEditNnc, buffer, position);
+    serializer.unpack(m_simulationConfig, buffer, position);
+    serializer.unpack(m_transMult, buffer, position);
+    serializer.unpack(m_faults, buffer, position);
+    serializer.unpack(m_title, buffer, position);
+}
+
+
+const FieldPropsManager& ParallelEclipseState::fieldProps() const
+{
+    if (!m_parProps)
+        assert(Dune::MPIHelper::getCollectiveCommunication().rank() == 0);
+
+    if (!m_parProps || Dune::MPIHelper::getCollectiveCommunication().size() == 1)
+        return this->EclipseState::fieldProps();
+
+    return m_fieldProps;
+}
+
+
+const FieldPropsManager& ParallelEclipseState::globalFieldProps() const
+{
+    assert(Dune::MPIHelper::getCollectiveCommunication().rank() == 0);
+    return this->EclipseState::globalFieldProps();
+}
+
+
+const EclipseGrid& ParallelEclipseState::getInputGrid() const
+{
+    assert(Dune::MPIHelper::getCollectiveCommunication().rank() == 0);
+    return this->EclipseState::getInputGrid();
+}
+
+
+void ParallelEclipseState::switchToGlobalProps()
+{
+    m_parProps = false;
+}
+
+
+void ParallelEclipseState::switchToDistributedProps()
+{
+    const auto& comm = Dune::MPIHelper::getCollectiveCommunication();
+    if (comm.size() == 1) // No need for the parallel frontend
+        return;
+
+    m_parProps = true;
+}
+
+void ParallelEclipseState::setupLocalProps(const std::vector<int>& localToGlobal)
+{
+#if HAVE_MPI
+    const auto& comm = Dune::MPIHelper::getCollectiveCommunication();
+    if (comm.rank() == 0) {
+        for (int i = 0; i < comm.size(); ++i) {
+            std::vector<int> l2gCell;
+            if (i == 0) {
+                for (const std::string& key : this->globalFieldProps().keys<int>()) {
```

Comment:
A bit much of code duplication. Maybe move the loop to a method taking functors that call the correct fieldProps member (e.g. `m_intProps`/`m_doubleProps`).

Reply:
okay, i did not want to spend much effort on code I will be replacing. but if it is required for merge, i'll get to it.